File: sdks/dotnet/sulfone-helium/Api/Resolver/Lambda.cs
using sulfone_helium.Domain.Core;
using sulfone_helium.Domain.Resolver;

namespace sulfone_helium.Api.Resolver;

public class LambdaResolver(Func<ResolverInput, Task<ResolverOutput>> f) : ICyanResolver
{
    public Task<ResolverOutput> Resolve(ResolverInput input)
    {
        return f(input);
    }
}

File: sdks/node/src/api/resolver/lambda.ts
import type { ResolverInput } from '../../domain/resolver/input.js';
import type { ResolverOutput } from '../../domain/resolver/output.js';
import type { ICyanResolver } from '../../domain/core/cyan_script.js';

type LambdaResolverFn = (i: ResolverInput) => Promise<ResolverOutput>;

class LambdaResolver implements ICyanResolver {
  private readonly _f: LambdaResolverFn;

  constructor(f: LambdaResolverFn) {
    this._f = f;
  }

  resolve(i: ResolverInput): Promise<ResolverOutput> {
    return this._f(i);
  }
}

export type { LambdaResolverFn };
export { LambdaResolver };
